src/socket.rs

```diff
@@ -370,6 +370,19 @@ impl Socket {
         self.inner.set_ttl(ttl)
     }
 
+    /// Sets the value for the `SO_MARK` option on this socket.
+    ///
+    /// This value sets the socket mark field for each packet sent through
+    /// this socket. Changing the mark can be used for mark-based routing
+    /// without netfilter or for packet filtering.
+    ///
+    /// This function is only available on Unix and requires the
+    /// `CAP_NET_ADMIN` capability.
+    #[cfg(target_os = "linux")]
+    pub fn set_mark(&self, mark: u32) -> io::Result<()> {
```

Comment:
I think `mark` should be `libc::c_int`, then we don't have to convert it in `sys/unix.rs`.

Reply:
Would that not push the cast into the application code? Using `c_int` is consistent with `set_ttl` etc.